uberenv.py

```diff
@@ -90,35 +90,64 @@ def parse_args():
                       action="store_true",
                       dest="install",
                       default=False,
-                      help="Install `package_name` instead of `uberenv_package_name`.")
+                      help="Install `package_name`, not just its dependencies.")
+
     # where to install
     parser.add_option("--prefix",
                       dest="prefix",
                       default="uberenv_libs",
                       help="destination directory")
+
     # what compiler to use
     parser.add_option("--spec",
                       dest="spec",
                       default=None,
                       help="spack compiler spec")
+
     # optional location of spack mirror
     parser.add_option("--mirror",
                       dest="mirror",
                       default=None,
                       help="spack mirror directory")
+
     # flag to create mirror
     parser.add_option("--create-mirror",
                       action="store_true",
                       dest="create_mirror",
                       default=False,
                       help="Create spack mirror")
+
+    # optional location of spack upstream
+    parser.add_option("--upstream",
+                      dest="upstream",
+                      default=None,
+                      help="add an external spack instance as upstream")
+
     # this option allows a user to explicitly to select a
     # group of spack settings files (compilers.yaml , packages.yaml)
     parser.add_option("--spack-config-dir",
                       dest="spack_config_dir",
                       default=None,
                       help="dir with spack settings files (compilers.yaml, packages.yaml, etc)")
 
+    # overrides package_name
+    parser.add_option("--package-name",
+                      dest="package_name",
+                      default=None,
+                      help="override the default package name")
+
+    # controls after which package phase spack should stop
+    parser.add_option("--package-final-phase",
```

Comment:
Valid phases are any phase defined in the Spack package. I could give typical examples.

Reply:
yes, the names of the phases come from the spack package and could be anything folks want

Reply:
Thanks.  

I'm not too familiar with this feature, which seems really useful. It seems like other ``uberenv`` users might have some trouble with this as well.

Would it be possible to add a parameter / command line arg to print those out for the requested package?  Or is it something that is easy to find by visual inspection of the file, so that won't be necessary?

Reply:
This is very recent, and a real improvement. 
Otherwise there are three methods:
- spack info <package> | grep -A 1 "Installation Phases" | tail -n 1
- if the phases are the defaults ones for this class of package (say CMakePackage), and then you may "know" those are "cmake", "build", "install" (not even certain), or check the documentation (https://spack.readthedocs.io/en/latest/build_systems.html).
- If the phases have been changed then you probably redefined the list in the packages (phases = ["hostconfig", "cmake", "build", "install"] ).